### ISSUE TITLE
Populate printed form with data

### ISSUE THREAD

**Issue by AlexZhidkov** [LegalX/PartWays#43]
File src\app\print-form\print-form.component.html contains form for printing out document, prepared for submitting to court. Data entered by user is stored in appData variable of the component. We need to populate the form with that data following this example:
```
    <div class="input-group">
    	<label>What is your family name as used now?</label>
    	<div class="fake-input">{{appData.applicant.familyName}}</div>
    </div>
    <div class="input-group">
    	<label>Given names?</label>
    	<div class="fake-input">{{appData.applicant.givenNames}}</div>
    </div>
```

**Comment by skusunam** [comment]
Let me know if you want me to send a PR for this?

**Comment by AlexZhidkov** [comment]
Yes, we need PR for this issue.

**Comment by skusunam** [comment]
I will send this out tomorrow (going to bed right now).

**Comment by CorreyL** [comment]
Hey Alex, is this what you're looking for?

![filledapplication](https://cloud.githubusercontent.com/assets/16601729/24281333/e4e0989e-1014-11e7-954d-e9dc6f322004.png)
![consentorder](https://cloud.githubusercontent.com/assets/16601729/24281334/e4e0de08-1014-11e7-8e3c-b508c621efcd.PNG)

Saw your post on Upwork. I'm a student learning AngularJS 2, still waiting to be approved on Upwork. Seemed like a fun challenge.

**Comment by AlexZhidkov** [comment]
@CorreyL What's the purpose of *ngIf there?

**Comment by CorreyL** [comment]
I'm subscribing to a FirebaseObjectObservable in print-form.component.ts, thus appData is not assigned immediately upon ngOnInit().

*ngIf ensures that Angular doesn't attempt to render the <div> until appData has been assigned.

**Comment by AlexZhidkov** [comment]
appData is resolved in resolver.

**Comment by CorreyL** [comment]
I opted to retrieve the object in a new function in party.resolver.service, and call that function in print-form.component.

I could probably move the new function to print-form.service.

**Comment by AlexZhidkov** [comment]
There's nothing needs to be done in service. Data retrieval already there and working. We won't give you this job, but if you want to learn Angular2 and prove yourself you can submit PR for any other issue here, I will review and gi

**Comment by AlexZhidkov** [comment]
And give you feedback. And we might consider you for another paid work.

**Comment by CorreyL** [comment]
Will do, appreciate the feedback =)

**Comment by unspike** [comment]
Hi, as i understood correctly you get pure html after clicking 'print' without  'values', right ?

**Comment by AlexZhidkov** [comment]
@unspike Yes. But HTML will have to be modified as well in cases where we need to iterate over array of data.

**Comment by paxibay** [comment]
import { Component, Input } from '@angular/core';
import { AppDataService, Applicant } from './services/appData-service';

@Component({
  selector: 'tag name goes here',
  styleUrls: [ 'css name goes here' ],
  template: `
    <div class="input-group">
    	<label>What is your family name as used now?</label>
    	<div class="fake-input">{{applicant.familyName}}</div>
    </div>
    <div class="input-group">
    	<label>Given names?</label>
    	<div class="fake-input">{{applicant.givenName}}</div>
    </div>
  `
})
export default class ApplicantComponent {
    private applicant: Applicant;

    // applicantId should ba passed from parent, here using default value
    @Input() applicantId: number = 0;  

    constructor(private appData: AppDataService) {
        this.applicant = this.appData.getApplicantById(applicantId);
    }
}

// put the following three blocks (Applicant, ProductService and applicants) 
// into 'services/appData-service' file, and then import it via:
// import { AppData, Applicant } from './services/appData-service';
export class Applicant {
    constructor(
        public id: number,
        public title: string,
        public price: number,
        public rating: number,
        public description: string,
        public categories: string[]) {
    }
}

export class AppDataService {
    getApplicantById(applicantId: number): Applicant {
        return applicants.find(a => a.id === applicantId);
    }
}

var applicants = [
    {
        "id": 0,
        "familyName": "familyName01",
        "givenName": "givenName01"
    },
    {
        "id": 1,
        "familyName": "familyName02",
        "givenName": "givenName02"
    },
    {
        "id": 2,
        "familyName": "familyName03",
        "givenName": "givenName03"
    },
    {
        "id": 3,
        "familyName": "familyName04",
        "givenName": "givenName04"
    }
];

**Comment by AlexZhidkov** [comment]
@paxibay we don't need service. It is already implemented. We need just HTML change.

**Comment by paxibay** [comment]
import { Component, Input } from '@angular/core';
import { AppDataService, Applicant } from './services/appData-service';

@Component({
  selector: 'tag name goes here',
  styleUrls: [ 'css name goes here' ],
  template: `
    <div class="input-group">
    	<label>What is your family name as used now?</label>
    	<div class="fake-input">{{applicant.familyName}}</div>
    </div>
    <div class="input-group">
    	<label>Given names?</label>
    	<div class="fake-input">{{applicant.givenName}}</div>
    </div>
  `
})
export default class ApplicantComponent {
    private applicant: Applicant;

    // applicantId should ba passed from parent, here using default value
    @Input() applicantId: number = 0;  

    constructor(private appData: AppDataService) {
        this.applicant = this.appData.getApplicantById(applicantId);
    }
}

**Comment by AlexZhidkov** [comment]
appData is resolved using resolver

**Comment by unspike** [comment]
i've checked your directuve it works as expected with rendered values... 
i guess problem is here `.data['application']`

**Comment by paxibay** [comment]
It's interesting. On my side, it works.

**Comment by unspike** [comment]
First of all try to use static values as @paxibay said

**Comment by paxibay** [comment]
Hi AlexZhidkov, I found you from upwork because I am thinking working as freelancer from home. Hope we could be friends.

**Comment by unspike** [comment]
Same as i.
About this issue i believe that u need to check resolver part, cuz other stuff works without troubles.

**Comment by AlexZhidkov** [comment]
Hi. If you're interested in this project, you can create PR for any of the small issues here. I will review your code and will give you feedback. If you're good we will consider you for paid job in the future.

**Comment by unspike** [comment]
> Hi. If you're interested in this project, you can create PR for any of the small issues here. I will review your code and will give you feedback. If you're good we will consider you for paid job in the future.
fixing issues is already part of job, so i can send proposal on upwork and u can ask whatever you need like on interview

**Comment by paxibay** [comment]
I want to ask a question what is appData.applicant? As I see, it should be a collection. So Angular doesn't know which applicant you want to select.

**Comment by AlexZhidkov** [comment]
You need to be logged in to see data. Login button on top right corner.

**Comment by unspike** [comment]
@paxibay 
`const data = this.af.database.object(`/application/${localStorage.getItem('applicationId')}`);
        return data.map((item) => {
            return item;
        }).first();`
this part should return a single value from collection based on 'applicationId'

**Comment by paxibay** [comment]
debug appData.applicant to see if it is a single object.

**Comment by paxibay** [comment]
Hi guys, I have to leave now, nice to see you here.

**Comment by paxibay** [comment]
I was wondering the IDE you used. There are so many IDEs which support using TypeScript, such as sublime text, webstorm, Visual Studio Code and even Visual Studio 2015/2017 standard version all sopport type checking at compile time. So before hard coded within template, just put the appData.applicant.familyName inside component, that's all.

**Comment by etrupja** [comment]
@AlexZhidkov I logged in and checked for the component you are saying.
On the solution I could not find a file named print-form.component.html

**Comment by etrupja** [comment]
@AlexZhidkov issue solved
![solution](https://cloud.githubusercontent.com/assets/20356633/24310623/f33c91d8-10d0-11e7-9b55-19da11ce9254.PNG)

**Comment by etrupja** [comment]
@AlexZhidkov changes that need to be done
![image](https://cloud.githubusercontent.com/assets/20356633/24310661/1616e582-10d1-11e7-843e-7306b5858293.png)

**Comment by unspike** [comment]
@etrupja lol

**Comment by etrupja** [comment]
@unspike ?

**Comment by unspike** [comment]
@etrupja as u can see on top issue related with data resolver, i believe that @AlexZhidkov fully understand that if it would be replaced (by static value) it will work, and main problem related with resolver and data-transport service.
so your sample will work - but its dosent acceptable answer i guess

**Comment by unspike** [comment]
@AlexZhidkov based on lates commits
u still need to figureout this part `this.route.snapshot.data['parties'];` in `PartyComponent`
but any way, just for solving issue of binding u can check this one :
```
    constructor(private route: ActivatedRoute, private af: AngularFire) {
        this.parties = {applicant:{}};
    }

    ngOnInit() {
        //this.parties = this.route.snapshot.data['parties'];
        this.partiesRef = this.af.database.object(`/application/${localStorage.getItem('applicationId')}`);
    }
```
and it will work
![image](https://cloud.githubusercontent.com/assets/636164/24312860/2fe71eac-10e3-11e7-8e06-50e662db1af0.png)

![image](https://cloud.githubusercontent.com/assets/636164/24312874/43efa400-10e3-11e7-8a14-e6e7a744ca94.png)

![image](https://cloud.githubusercontent.com/assets/636164/24312888/50d9f062-10e3-11e7-85da-00d280fdf014.png)

**Comment by unspike** [comment]
@AlexZhidkov hey, help wasnt accepted and taks (on upwork) still opened?

**Comment by AlexZhidkov** [comment]
Hi @unspike, thank you for trying to help, but you haven't quite understood requirements for this work. Job on UpWork is offered to another developer.
